caps/capabilities.go

```diff
@@ -137,6 +161,24 @@ func (c Capability) String() string {
 	return c.Name
 }
 
+// TypeNames returns all type names in the repository in lexicographical order.
+func (r *Repository) TypeNames() []string {
+	r.Lock()
+	defer r.Unlock()
+
+	types := make([]string, len(r.types))
+	for i, t := range r.types {
+		types[i] = t.String()
+	}
+	sort.Strings(types)
+	return types
+}
+
+// String representation of a capability type
+func (t Type) String() string {
```

Comment:
This seems a bit lost. Should probably be defined next to the type itself.

Also, let's please be consistent about the comment style, following the upstream convention:

// String returns a string representation for the capability type.